### ISSUE TITLE
Crash when updating/re-deploy with fog

### ISSUE THREAD

**Issue by mkasu** [nanoc/nanoc#718]
Previously I've been deploying my blog with rsync without trouble so far. Today I switched to fog to deploy my blog to AWS.

The initial upload went through without problems. However, I can't update my blog anymore, as any subsequent call of "nanoc deploy" crashes.

Am I doing anything wrongly?

nanoc.yaml
```
deploy:
  default:
    kind: fog
    provider: aws
    region: eu-west-1
    bucket: mkasu.org
    aws_access_key_id: ....
    aws_secret_access_key: ....
    path_style: true
```

crash.log
```
Crashlog created at 2015-11-06 22:15:34 +0100

===== MESSAGE:

NoMethodError: undefined method `all' for #<Array:0x007fa784918218>

===== COMPILATION STACK:

  (empty)

===== STACK TRACE:

  0. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/extra/deployers/fog.rb:72:in `run'
  1. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/cli/commands/deploy.rb:85:in `run'
  2. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/cli/command_runner.rb:12:in `block in call'
  3. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/cli/error_handler.rb:64:in `handle_while'
  4. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/cli/error_handler.rb:22:in `handle_while'
  5. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/cli/command_runner.rb:11:in `call'
  6. /usr/local/lib/ruby/gems/2.2.0/gems/cri-2.7.0/lib/cri/command_dsl.rb:233:in `block in runner'
  7. /usr/local/lib/ruby/gems/2.2.0/gems/cri-2.7.0/lib/cri/command.rb:294:in `call'
  8. /usr/local/lib/ruby/gems/2.2.0/gems/cri-2.7.0/lib/cri/command.rb:294:in `run_this'
  9. /usr/local/lib/ruby/gems/2.2.0/gems/cri-2.7.0/lib/cri/command.rb:247:in `run'
  10. /usr/local/lib/ruby/gems/2.2.0/gems/cri-2.7.0/lib/cri/command.rb:260:in `run'
  11. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/cli.rb:53:in `block in run'
  12. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/cli/error_handler.rb:64:in `handle_while'
  13. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/cli/error_handler.rb:22:in `handle_while'
  14. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib/nanoc/cli.rb:51:in `run'
  15. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/bin/nanoc:9:in `<top (required)>'
  16. /usr/local/bin/nanoc:23:in `load'

===== VERSION INFORMATION:

Nanoc 4.0.0rc3 © 2007-2015 Denis Defreyne.
Running ruby 2.2.3 (2015-08-18) on x86_64-darwin15 with RubyGems 2.4.5.1.

===== SYSTEM INFORMATION:

Darwin Marcs-MacBook-Pro.local 15.0.0 Darwin Kernel Version 15.0.0: Sat Sep 19 15:53:46 PDT 2015; root:xnu-3247.10.11~1/RELEASE_X86_64 x86_64

===== INSTALLED GEMS:

  CFPropertyList 2.3.2
  actionpack 4.2.4
  actionview 4.2.4
  activesupport 4.2.4
  adsf 1.2.0
  builder 3.2.2
  bundler 1.10.6
  chunky_png 1.3.5
  coderay 1.1.0
  colored 1.2
  compass 1.0.3
  compass-core 1.0.3
  compass-import-once 1.0.5
  cri 2.7.0
  erubis 2.7.0
  excon 0.45.4
  execjs 2.6.0
  ffi 1.9.10
  fission 0.5.0
  fog 1.35.0
  fog-aliyun 0.0.10
  fog-atmos 0.1.0
  fog-aws 0.7.6
  fog-brightbox 0.9.0
  fog-core 1.32.1
  fog-dynect 0.0.2
  fog-ecloud 0.3.0
  fog-google 0.1.1
  fog-json 1.0.2
  fog-local 0.2.1
  fog-powerdns 0.1.1
  fog-profitbricks 0.0.5
  fog-radosgw 0.0.4
  fog-riakcs 0.1.0
  fog-sakuracloud 1.3.3
  fog-serverlove 0.1.2
  fog-softlayer 1.0.2
  fog-storm_on_demand 0.1.1
  fog-terremark 0.1.0
  fog-vmfusion 0.1.0
  fog-voxel 0.1.0
  fog-xenserver 0.2.2
  fog-xml 0.1.2
  formatador 0.2.5
  foundation-rails 5.5.3.2
  guard 2.13.0
  guard-compat 1.2.1
  guard-nanoc 2.0.0b1
  haml 4.0.7
  i18n 0.7.0
  inflecto 0.0.2
  ipaddress 0.8.0
  json 1.8.3
  kramdown 1.9.0
  listen 3.0.3
  loofah 2.0.3
  lumberjack 1.0.9
  method_source 0.8.2
  mime-types 2.6.2
  mini_portile 0.6.2
  minitest 5.8.2
  multi_json 1.11.2
  nanoc 4.0.0rc3
  nenv 0.2.0
  net-scp 1.2.1
  net-ssh 3.0.1
  nokogiri 1.6.6.2
  notiffany 0.0.8
  pry 0.10.3
  rack 1.6.4
  rack-test 0.6.3
  rails-deprecated_sanitizer 1.0.3
  rails-dom-testing 1.0.7
  rails-html-sanitizer 1.0.2
  railties 4.2.4
  rainpress 1.0
  rake 10.4.2
  rb-fsevent 0.9.6
  rb-inotify 0.9.5
  sass 3.4.19
  shellany 0.0.1
  slop 3.6.0
  thor 0.19.1
  thread_safe 0.3.5
  tilt 2.0.1
  tzinfo 1.2.2
  uglifier 2.7.2
  xml-simple 1.1.5

===== ENVIRONMENT:

Apple_PubSub_Socket_Render => "/private/tmp/com.apple.launchd.JArVuU3nKO/Render"
BUNDLE_BIN_PATH => "/usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.10.6/bin/bundle"
BUNDLE_GEMFILE => "/Users/mkastner/blog/Gemfile"
CLICOLOR => "1"
COLORTERM => "yes"
DISPLAY => "/private/tmp/com.apple.launchd.jodsQWxmIQ/org.macosforge.xquartz:0"
EDITOR => "vim"
GEM_HOME => "/usr/local/lib/ruby/gems/2.2.0"
GEM_PATH => "/usr/local/lib/ruby/gems/2.2.0:/Users/mkastner/.gem/ruby/2.2.0:/usr/local/Cellar/ruby/2.2.3/lib/ruby/gems/2.2.0"
HOME => "/Users/mkastner"
ITERM_PROFILE => "Default"
ITERM_SESSION_ID => "w0t4p1:CE753086-BD1B-45C1-A97D-BDD7995257CA"
LC_CTYPE => "UTF-8"
LESS_TERMCAP_mb => "\e[01;31m"
LESS_TERMCAP_md => "\e[01;31m"
LESS_TERMCAP_me => "\e[0m"
LESS_TERMCAP_se => "\e[0m"
LESS_TERMCAP_so => "\e[01;44;33m"
LESS_TERMCAP_ue => "\e[0m"
LESS_TERMCAP_us => "\e[01;32m"
LOGNAME => "mkastner"
MAIL => "/var/mail/mkastner"
MANPATH => "/usr/local/lib/ruby/gems/2.2.0/gems/kramdown-1.9.0/man"
OLDPWD => "/Users/mkastner"
PAGER => "less"
PATH => "/usr/local/lib/ruby/gems/2.2.0/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/TeX/texbin"
PWD => "/Users/mkastner/blog"
RUBYLIB => "/usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.10.6/lib"
RUBYOPT => "-rbundler/setup"
SHELL => "/usr/local/bin/zsh"
SHLVL => "1"
SSH_AUTH_SOCK => "/private/tmp/com.apple.launchd.ZKXrDDAjag/Listeners"
TERM => "xterm-256color"
TERM_PROGRAM => "iTerm.app"
TMPDIR => "/var/folders/mh/4lv_h_q920q3hjt9wzlrfnkh0000gn/T/"
USER => "mkastner"
XPC_FLAGS => "0x0"
XPC_SERVICE_NAME => "0"
_ => "/usr/local/bin/bundle"
__CF_USER_TEXT_ENCODING => "0x1F5:0x0:0x0"

===== GEMFILE.LOCK:

GIT
  remote: git://github.com/guard/guard-nanoc.git
  revision: 7a44c90333a4077c63a710bbf36216299929d27d
  specs:
    guard-nanoc (2.0.0b1)
      guard (~> 2.8)
      guard-compat (~> 1.0)
      nanoc (~> 4.0.0b4)

GEM
  remote: https://rubygems.org/
  specs:
    CFPropertyList (2.3.2)
    actionpack (4.2.4)
      actionview (= 4.2.4)
      activesupport (= 4.2.4)
      rack (~> 1.6)
      rack-test (~> 0.6.2)
      rails-dom-testing (~> 1.0, >= 1.0.5)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (4.2.4)
      activesupport (= 4.2.4)
      builder (~> 3.1)
      erubis (~> 2.7.0)
      rails-dom-testing (~> 1.0, >= 1.0.5)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    activesupport (4.2.4)
      i18n (~> 0.7)
      json (~> 1.7, >= 1.7.7)
      minitest (~> 5.1)
      thread_safe (~> 0.3, >= 0.3.4)
      tzinfo (~> 1.1)
    adsf (1.2.0)
      rack (>= 1.0.0)
    allison (2.0.3)
    builder (3.2.2)
    chunky_png (1.3.5)
    coderay (1.1.0)
    colored (1.2)
    compass (1.0.3)
      chunky_png (~> 1.2)
      compass-core (~> 1.0.2)
      compass-import-once (~> 1.0.5)
      rb-fsevent (>= 0.9.3)
      rb-inotify (>= 0.9)
      sass (>= 3.3.13, < 3.5)
    compass-core (1.0.3)
      multi_json (~> 1.0)
      sass (>= 3.3.0, < 3.5)
    compass-import-once (1.0.5)
      sass (>= 3.2, < 3.5)
    cri (2.7.0)
      colored (~> 1.2)
    echoe (4.6.6)
      allison (>= 2.0.3)
      rake (>= 0.9.2)
      rdoc (>= 2.5.11)
      rubyforge (>= 2.0.4)
    erubis (2.7.0)
    excon (0.45.4)
    execjs (2.6.0)
    ffi (1.9.10)
    fission (0.5.0)
      CFPropertyList (~> 2.2)
    fog (1.35.0)
      fog-aliyun
      fog-atmos
      fog-aws (>= 0.6.0)
      fog-brightbox (~> 0.4)
      fog-core (~> 1.32)
      fog-dynect (~> 0.0.2)
      fog-ecloud (~> 0.1)
      fog-google (>= 0.1.1)
      fog-json
      fog-local
      fog-powerdns (>= 0.1.1)
      fog-profitbricks
      fog-radosgw (>= 0.0.2)
      fog-riakcs
      fog-sakuracloud (>= 0.0.4)
      fog-serverlove
      fog-softlayer
      fog-storm_on_demand
      fog-terremark
      fog-vmfusion
      fog-voxel
      fog-xenserver
      fog-xml (~> 0.1.1)
      ipaddress (~> 0.5)
      nokogiri
    fog-aliyun (0.0.10)
      fog-core (~> 1.27)
      fog-json (~> 1.0)
      ipaddress (~> 0.8)
      xml-simple (~> 1.1)
    fog-atmos (0.1.0)
      fog-core
      fog-xml
    fog-aws (0.7.6)
      fog-core (~> 1.27)
      fog-json (~> 1.0)
      fog-xml (~> 0.1)
      ipaddress (~> 0.8)
    fog-brightbox (0.9.0)
      fog-core (~> 1.22)
      fog-json
      inflecto (~> 0.0.2)
    fog-core (1.32.1)
      builder
      excon (~> 0.45)
      formatador (~> 0.2)
      mime-types
      net-scp (~> 1.1)
      net-ssh (>= 2.1.3)
    fog-dynect (0.0.2)
      fog-core
      fog-json
      fog-xml
    fog-ecloud (0.3.0)
      fog-core
      fog-xml
    fog-google (0.1.1)
      fog-core
      fog-json
      fog-xml
    fog-json (1.0.2)
      fog-core (~> 1.0)
      multi_json (~> 1.10)
    fog-local (0.2.1)
      fog-core (~> 1.27)
    fog-powerdns (0.1.1)
      fog-core (~> 1.27)
      fog-json (~> 1.0)
      fog-xml (~> 0.1)
    fog-profitbricks (0.0.5)
      fog-core
      fog-xml
      nokogiri
    fog-radosgw (0.0.4)
      fog-core (>= 1.21.0)
      fog-json
      fog-xml (>= 0.0.1)
    fog-riakcs (0.1.0)
      fog-core
      fog-json
      fog-xml
    fog-sakuracloud (1.3.3)
      fog-core
      fog-json
    fog-serverlove (0.1.2)
      fog-core
      fog-json
    fog-softlayer (1.0.2)
      fog-core
      fog-json
    fog-storm_on_demand (0.1.1)
      fog-core
      fog-json
    fog-terremark (0.1.0)
      fog-core
      fog-xml
    fog-vmfusion (0.1.0)
      fission
      fog-core
    fog-voxel (0.1.0)
      fog-core
      fog-xml
    fog-xenserver (0.2.2)
      fog-core
      fog-xml
    fog-xml (0.1.2)
      fog-core
      nokogiri (~> 1.5, >= 1.5.11)
    formatador (0.2.5)
    foundation-rails (5.5.3.2)
      railties (>= 3.1.0)
      sass (>= 3.3.0, < 3.5)
    guard (2.13.0)
      formatador (>= 0.2.4)
      listen (>= 2.7, <= 4.0)
      lumberjack (~> 1.0)
      nenv (~> 0.1)
      notiffany (~> 0.0)
      pry (>= 0.9.12)
      shellany (~> 0.0)
      thor (>= 0.18.1)
    guard-compat (1.2.1)
    haml (4.0.7)
      tilt
    i18n (0.7.0)
    inflecto (0.0.2)
    ipaddress (0.8.0)
    json (1.8.3)
    json_pure (1.8.3)
    kramdown (1.9.0)
    listen (3.0.3)
      rb-fsevent (>= 0.9.3)
      rb-inotify (>= 0.9)
    loofah (2.0.3)
      nokogiri (>= 1.5.9)
    lumberjack (1.0.9)
    method_source (0.8.2)
    mime-types (2.6.2)
    mini_portile (0.6.2)
    minitest (5.8.2)
    multi_json (1.11.2)
    nanoc (4.0.0rc3)
      cri (~> 2.3)
    nenv (0.2.0)
    net-scp (1.2.1)
      net-ssh (>= 2.6.5)
    net-ssh (3.0.1)
    nokogiri (1.6.6.2)
      mini_portile (~> 0.6.0)
    notiffany (0.0.8)
      nenv (~> 0.1)
      shellany (~> 0.0)
    pry (0.10.3)
      coderay (~> 1.1.0)
      method_source (~> 0.8.1)
      slop (~> 3.4)
    rack (1.6.4)
    rack-test (0.6.3)
      rack (>= 1.0)
    rails-deprecated_sanitizer (1.0.3)
      activesupport (>= 4.2.0.alpha)
    rails-dom-testing (1.0.7)
      activesupport (>= 4.2.0.beta, < 5.0)
      nokogiri (~> 1.6.0)
      rails-deprecated_sanitizer (>= 1.0.1)
    rails-html-sanitizer (1.0.2)
      loofah (~> 2.0)
    railties (4.2.4)
      actionpack (= 4.2.4)
      activesupport (= 4.2.4)
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rainpress (1.0)
      echoe
    rake (10.4.2)
    rb-fsevent (0.9.6)
    rb-inotify (0.9.5)
      ffi (>= 0.5.0)
    rdoc (4.2.0)
    rubyforge (2.0.4)
      json_pure (>= 1.1.7)
    sass (3.4.19)
    shellany (0.0.1)
    slop (3.6.0)
    thor (0.19.1)
    thread_safe (0.3.5)
    tilt (2.0.1)
    tzinfo (1.2.2)
      thread_safe (~> 0.1)
    uglifier (2.7.2)
      execjs (>= 0.3.0)
      json (>= 1.8.0)
    xml-simple (1.1.5)

PLATFORMS
  ruby

DEPENDENCIES
  adsf
  compass
  fog
  foundation-rails
  guard
  guard-nanoc!
  haml
  kramdown
  nanoc (~> 4.0.0rc3)
  rainpress
  uglifier

BUNDLED WITH
   1.10.6

===== LOAD PATHS:

  0. /usr/local/lib/ruby/gems/2.2.0/gems/uglifier-2.7.2/lib
  1. /usr/local/lib/ruby/gems/2.2.0/gems/rainpress-1.0/lib
  2. /usr/local/lib/ruby/gems/2.2.0/gems/kramdown-1.9.0/lib
  3. /usr/local/lib/ruby/gems/2.2.0/gems/haml-4.0.7/lib
  4. /usr/local/lib/ruby/gems/2.2.0/gems/tilt-2.0.1/lib
  5. /Users/mkastner/.bundler/ruby/2.2.0/guard-nanoc-7a44c90333a4/lib
  6. /usr/local/lib/ruby/gems/2.2.0/gems/nanoc-4.0.0rc3/lib
  7. /usr/local/lib/ruby/gems/2.2.0/gems/guard-compat-1.2.1/lib
  8. /usr/local/lib/ruby/gems/2.2.0/gems/guard-2.13.0/lib
  9. /usr/local/lib/ruby/gems/2.2.0/gems/pry-0.10.3/lib
  10. /usr/local/lib/ruby/gems/2.2.0/gems/slop-3.6.0/lib
  11. /usr/local/lib/ruby/gems/2.2.0/gems/method_source-0.8.2/lib
  12. /usr/local/lib/ruby/gems/2.2.0/gems/notiffany-0.0.8/lib
  13. /usr/local/lib/ruby/gems/2.2.0/gems/shellany-0.0.1/lib
  14. /usr/local/lib/ruby/gems/2.2.0/gems/nenv-0.2.0/lib
  15. /usr/local/lib/ruby/gems/2.2.0/gems/lumberjack-1.0.9/lib
  16. /usr/local/lib/ruby/gems/2.2.0/gems/listen-3.0.3/lib
  17. /usr/local/lib/ruby/gems/2.2.0/gems/foundation-rails-5.5.3.2/lib
  18. /usr/local/lib/ruby/gems/2.2.0/gems/railties-4.2.4/lib
  19. /usr/local/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib
  20. /usr/local/lib/ruby/gems/2.2.0/gems/fog-1.35.0/lib
  21. /usr/local/lib/ruby/gems/2.2.0/gems/fog-xenserver-0.2.2/lib
  22. /usr/local/lib/ruby/gems/2.2.0/gems/fog-voxel-0.1.0/lib
  23. /usr/local/lib/ruby/gems/2.2.0/gems/fog-vmfusion-0.1.0/lib
  24. /usr/local/lib/ruby/gems/2.2.0/gems/fog-terremark-0.1.0/lib
  25. /usr/local/lib/ruby/gems/2.2.0/gems/fog-storm_on_demand-0.1.1/lib
  26. /usr/local/lib/ruby/gems/2.2.0/gems/fog-softlayer-1.0.2/lib
  27. /usr/local/lib/ruby/gems/2.2.0/gems/fog-serverlove-0.1.2/lib
  28. /usr/local/lib/ruby/gems/2.2.0/gems/fog-sakuracloud-1.3.3/lib
  29. /usr/local/lib/ruby/gems/2.2.0/gems/fog-riakcs-0.1.0/lib
  30. /usr/local/lib/ruby/gems/2.2.0/gems/fog-radosgw-0.0.4/lib
  31. /usr/local/lib/ruby/gems/2.2.0/gems/fog-profitbricks-0.0.5/lib
  32. /usr/local/lib/ruby/gems/2.2.0/gems/fog-powerdns-0.1.1/lib
  33. /usr/local/lib/ruby/gems/2.2.0/gems/fog-local-0.2.1/lib
  34. /usr/local/lib/ruby/gems/2.2.0/gems/fog-google-0.1.1/lib
  35. /usr/local/lib/ruby/gems/2.2.0/gems/fog-ecloud-0.3.0/lib
  36. /usr/local/lib/ruby/gems/2.2.0/gems/fog-dynect-0.0.2/lib
  37. /usr/local/lib/ruby/gems/2.2.0/gems/fog-brightbox-0.9.0/lib
  38. /usr/local/lib/ruby/gems/2.2.0/gems/inflecto-0.0.2/lib
  39. /usr/local/lib/ruby/gems/2.2.0/gems/fog-aws-0.7.6/lib
  40. /usr/local/lib/ruby/gems/2.2.0/gems/fog-atmos-0.1.0/lib
  41. /usr/local/lib/ruby/gems/2.2.0/gems/fog-xml-0.1.2/lib
  42. /usr/local/lib/ruby/gems/2.2.0/gems/fog-aliyun-0.0.10/lib
  43. /usr/local/lib/ruby/gems/2.2.0/gems/xml-simple-1.1.5/lib
  44. /usr/local/lib/ruby/gems/2.2.0/gems/ipaddress-0.8.0/lib
  45. /usr/local/lib/ruby/gems/2.2.0/gems/fog-json-1.0.2/lib
  46. /usr/local/lib/ruby/gems/2.2.0/gems/fog-core-1.32.1/lib
  47. /usr/local/lib/ruby/gems/2.2.0/gems/net-scp-1.2.1/lib
  48. /usr/local/lib/ruby/gems/2.2.0/gems/net-ssh-3.0.1/lib
  49. /usr/local/lib/ruby/gems/2.2.0/gems/mime-types-2.6.2/lib
  50. /usr/local/lib/ruby/gems/2.2.0/gems/formatador-0.2.5/lib
  51. /usr/local/lib/ruby/gems/2.2.0/gems/fission-0.5.0/lib
  52. /usr/local/lib/ruby/gems/2.2.0/gems/execjs-2.6.0/lib
  53. /usr/local/lib/ruby/gems/2.2.0/gems/excon-0.45.4/lib
  54. /usr/local/lib/ruby/gems/2.2.0/gems/cri-2.7.0/lib
  55. /usr/local/lib/ruby/gems/2.2.0/gems/compass-1.0.3/lib
  56. /usr/local/lib/ruby/gems/2.2.0/gems/rb-inotify-0.9.5/lib
  57. /usr/local/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-15/2.2.0/ffi-1.9.10
  58. /usr/local/lib/ruby/gems/2.2.0/gems/ffi-1.9.10/lib
  59. /usr/local/lib/ruby/gems/2.2.0/gems/rb-fsevent-0.9.6/lib
  60. /usr/local/lib/ruby/gems/2.2.0/gems/compass-import-once-1.0.5/lib
  61. /usr/local/lib/ruby/gems/2.2.0/gems/compass-core-1.0.3/lib
  62. /usr/local/lib/ruby/gems/2.2.0/gems/sass-3.4.19/lib
  63. /usr/local/lib/ruby/gems/2.2.0/gems/multi_json-1.11.2/lib
  64. /usr/local/lib/ruby/gems/2.2.0/gems/colored-1.2/lib
  65. /usr/local/lib/ruby/gems/2.2.0/gems/coderay-1.1.0/lib
  66. /usr/local/lib/ruby/gems/2.2.0/gems/chunky_png-1.3.5/lib
  67. /usr/local/lib/ruby/gems/2.2.0/gems/adsf-1.2.0/lib
  68. /usr/local/lib/ruby/gems/2.2.0/gems/actionpack-4.2.4/lib
  69. /usr/local/lib/ruby/gems/2.2.0/gems/rack-test-0.6.3/lib
  70. /usr/local/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib
  71. /usr/local/lib/ruby/gems/2.2.0/gems/actionview-4.2.4/lib
  72. /usr/local/lib/ruby/gems/2.2.0/gems/rails-html-sanitizer-1.0.2/lib
  73. /usr/local/lib/ruby/gems/2.2.0/gems/loofah-2.0.3/lib
  74. /usr/local/lib/ruby/gems/2.2.0/gems/rails-dom-testing-1.0.7/lib
  75. /usr/local/lib/ruby/gems/2.2.0/gems/rails-deprecated_sanitizer-1.0.3/lib
  76. /usr/local/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-15/2.2.0/nokogiri-1.6.6.2
  77. /usr/local/lib/ruby/gems/2.2.0/gems/nokogiri-1.6.6.2/lib
  78. /usr/local/lib/ruby/gems/2.2.0/gems/mini_portile-0.6.2/lib
  79. /usr/local/lib/ruby/gems/2.2.0/gems/erubis-2.7.0/lib
  80. /usr/local/lib/ruby/gems/2.2.0/gems/builder-3.2.2/lib
  81. /usr/local/lib/ruby/gems/2.2.0/gems/activesupport-4.2.4/lib
  82. /usr/local/lib/ruby/gems/2.2.0/gems/tzinfo-1.2.2/lib
  83. /usr/local/lib/ruby/gems/2.2.0/gems/thread_safe-0.3.5/lib
  84. /usr/local/lib/ruby/gems/2.2.0/gems/minitest-5.8.2/lib
  85. /usr/local/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-15/2.2.0/json-1.8.3
  86. /usr/local/lib/ruby/gems/2.2.0/gems/json-1.8.3/lib
  87. /usr/local/lib/ruby/gems/2.2.0/gems/i18n-0.7.0/lib
  88. /usr/local/lib/ruby/gems/2.2.0/gems/CFPropertyList-2.3.2/lib
  89. /usr/local/lib/ruby/gems/2.2.0/gems/rake-10.4.2/lib
  90. /usr/local/lib/ruby/gems/2.2.0/gems/bundler-1.10.6/lib
  91. /usr/local/lib/ruby/site_ruby/2.2.0
  92. /usr/local/lib/ruby/site_ruby/2.2.0/x86_64-darwin15
  93. /usr/local/lib/ruby/site_ruby
  94. /usr/local/lib/ruby/vendor_ruby/2.2.0
  95. /usr/local/lib/ruby/vendor_ruby/2.2.0/x86_64-darwin15
  96. /usr/local/lib/ruby/vendor_ruby
  97. /usr/local/Cellar/ruby/2.2.3/lib/ruby/2.2.0
  98. /usr/local/Cellar/ruby/2.2.3/lib/ruby/2.2.0/x86_64-darwin15
```

**Comment by ddfreyne** [comment]
I think this is a bug in Nanoc. I assume an instance of `Fog::Collection` is expected, but it turns out to be an `Array`, possibly due to the `files += set`.

Replacing

```ruby
keys_to_destroy = files.all.map(&:key)
```

with

```ruby
keys_to_destroy = files.map(&:key)
```

(i.e. removing the call to `#all`) might do the trick. After all, `files` is an instance of an `Array` subclass, so the `#all` shouldn’t be needed. Can you verify whether this works for you?

(I find that fog code is a little tricky to test.)

**Comment by mkasu** [comment]
Thank you for the quick response. I tried your patch and it seems to work splendidly. My update went through without trouble.

As a side note some "feature suggestion": Maybe a verbose flag would be useful for bigger deployments. The "Uploading local files" step can take a long of time. It'd be useful to see the current progress. Locally, I just added a "puts file_path" in the upload-loop, which works well enough for me, but a flag would probably be a better solution for upstream.
